contracts/RandomBeaconHistory.cdc

```diff
@@ -143,30 +127,28 @@ access(all) contract RandomBeaconHistory {
         //
         /// gaps only occur in the rare event of a system transaction failure. In this case, entries are still
         /// filled using a source not known at the time of block execution, which guarantees unpredicatability.
-        access(contract) fun backfill(randomSource: [UInt8]) {
-
-            let correctIndex = self.findGapAndReturnCorrectIndex()
-            var arrayLength = UInt64(RandomBeaconHistory.randomSourceHistory.length)
-            // optional optimization for the happy common path: if no gaps are detected,
+        access(contract) fun backfill(randomSource: [UInt8], currentArrayLength: UInt64, correctIndex: UInt64) {
```

Comment:
My preference for this interface is to feed the random source and gather the provided values directly from contract fields, especially since this method lacks conditional checks. The perceived risk here is that the method can be called with arbitrary values which are not guaranteed to be limited to the current block height.

IMO the use of `findGapAndReturnCorrectIndex()` in conjunction with `backfill(randomSource: [UInt8])` corrects gaps limited by the current block height with as far as I can tell a single duplicate call to `RandomBeaconHistory.randomSourceHistory.length`, which is cheap constant time.

Reply:
oh my bad I didn't mean to overwrite anything, the commit https://github.com/onflow/flow-core-contracts/pull/416/commits/9e5d257092612bdc431372fec17903e067a2a1ae purpose was only to address Bastian's comments, but I think there is a mess up in my local branches and I force-pushed, let me clean that up.

Reply:
It should be fixed now, commit https://github.com/onflow/flow-core-contracts/pull/416/commits/75d3500157010ad01bca07d1ad0bde0ef50d6bfc addresses Bastian's comments and shouldn't overwrite other things.